### DISCUSSION TITLE
Choice for VM or LXC?

### DISCUSSION THREAD

**Discussion by JustTooKrul** [tteck/Proxmox#2677] [Help Needed]
I know the scripts aren't mean to be used this way, but the application installation scripts are quite good and do a lot of nice housekeeping that would be great to have access to inside either a VM or a LXC. I tried "rolling my own" a few times and it caused disasters that made me need to re-do the VM entirely... (As an aside, how about giving an option to specify the disk size when creating a new VM? Little did I know 2G was going to--in short order--start "disk drive full" errors when installing Docker!).

Just a suggestion, but obviously all the work on these is much appreciated and I hate "looking a gift horse in the mouth" ... !

**Comment by tteck** [comment]
Why would someone choose a VM over an LXC container when LXC containers offer advantages in terms of performance and resource efficiency (other than cluster migration)?

Reading the linked instructions after creating an OS VM is very helpful.
- Debian https://github.com/tteck/Proxmox/discussions/1988
- Ubuntu https://github.com/tteck/Proxmox/discussions/2072

**Reply by JustTooKrul** [reply]
I had a _**ton**_ of problems setting up Docker as an LXC container and getting my iGPU to pass through correctly along with getting NFS working, especially given the restricted permissions... Giving the LXC container elevated permissions would both defeat the purpose and undo some of the point of using a LXC container to begin with.

Also, expanding storage and making some other tweaks are far less straightforward on a LXC container.

**Reply by tteck** [reply]
> would be great to have access to inside either a VM or a LXC

I have no plans to incorporate such a feature.  

> Giving the LXC container elevated permissions would both defeat the purpose and undo some of the point of using a LXC container to begin with.

I don't understand this statement.

> expanding storage and making some other tweaks are far less straightforward on a LXC container.

Many would suggest this is easier utilizing a LXC container.

Thanks for your suggestion/input on this.
